refract-spec.md

```diff
@@ -137,9 +139,490 @@ This is how it would expressed in the compact version.
+- `content` (array[Property Element], required)
+
+#### Examples
+
+In JSON, an example of an object is:
+
+```json
+{
+  "foo": "bar"
+}
+```
+
+In Refract, this object is expanded to a `Object Element`.
+
+```json
+{
+  "element": "object",
+  "attributes": {},
+  "content": [
+    {
+      "element": "property",
+      "attributes": {
+        "name": "foo"
+      },
+      "content": {
+        "element": "string",
+        "attributes": {},
+        "content": "bar"
+      }
+    }
+  ]
+}
+```
+
+## Referencing Elements
+
+The referencing elements are provided as part of the base specification for the purpose of identifying and referencing elements.
+
+### ID (Element)
+
+The `id` element is a way to identify an element in a document. IDs MUST be unique throughout a document.
+
+#### Example
+
+```json
+{
+  "element": "string",
+  "attributes": {
+    "id": "foo"
+  },
+  "content": "bar"
+}
+```
+
+### Ref (Element)
+
+The `ref` element MAY be used to reference elements in remote documents or elements in the local document.
+
+1. If the content is an absolute URL with a fragment identifier, the fragment identifer references the ID in the corresponding document.
+1. If the content is a relative fragment identifier, it references an element found in the local document
+1. If there is no fragment identifier, the reference is to the entire document.
+1. If there are multiple elements in a document with the same ID, the first element with the matching value SHOULD be used.
+
+The `ref` element transcludes the contents of the element into the document in which it is referenced.
+
+#### Properties
+
+- `element` ref (string, fixed)
+- `content` (string) - A URL
+
+#### Examples
+
+Elements MAY be referenced in remove or local documents.
+
+##### Referencing Remote Element
+
+```json
+{
+  "element": "ref",
+  "content": "http://example.com/document#foo"
+}
+```
+
+##### Referencing Local Elements
+
+```json
+{
+  "element": "ref",
+  "content": "#foo"
+}
+```
+
+## Namespacing
+
+Namespaces provide a way to include element definitions and constraints into another document. The following rules and constraints apply to namespaces.
+
+1. If there is a conflict, the namespaced element defined or referenced last MUST take precedence
+1. Types defined within the local namespace MUST take precedence over referenced namespaces
+1. Prefixed namespaces are only accessible through using a prefix
+1. Namespaces apply to the element in which they are introduced and all child elements
+1. Namespaces MUST not include the elements nor the data in the elements referenced, but rather include the constraints and conditions around the types defined within that namespace
+
+### Namespaces (Element)
+
+The `namespaces` element is a list of Namespace elements.
+
+#### Properties
+
+- `element` namespaces (string, fixed)
+- `content` (array[Namespace])
+
+### Namespace (enum)
+
+#### Members
+
+- (string) - A resolvable URL to a namespace
```

Comment:
I would say this is missing an explanation between `A resolvable URL to a namespace` and `An href is a resolvable URL to a namespace`. I assume those are the same, the later (in the object) is to define a prefix. 

But please elaborate.